warehouse/forklift/legacy.py

```diff
@@ -1169,28 +1200,33 @@ def file_upload(request):
                     "requires_python",
                 }
             },
-            uploader=request.user,
+            uploader=request.user if request.user else None,
             uploaded_via=request.user_agent,
         )
         request.db.add(release)
-        # TODO: This should be handled by some sort of database trigger or
-        #       a SQLAlchemy hook or the like instead of doing it inline in
-        #       this view.
-        request.db.add(
-            JournalEntry(
-                name=release.project.name,
-                version=release.version,
-                action="new release",
-                submitted_by=request.user,
-                submitted_from=request.remote_addr,
+
+        if request.user:
+            # TODO: This should be handled by some sort of database trigger or
+            #       a SQLAlchemy hook or the like instead of doing it inline in
+            #       this view.
+            request.db.add(
+                JournalEntry(
+                    name=release.project.name,
+                    version=release.version,
+                    action="new release",
+                    submitted_by=request.user,
+                    submitted_from=request.remote_addr,
```

Comment:
NTS: I'll relax `JournalEntry.submitted_by` to make it nullable, so that we can re-enable this journal entry for all request identities.